src/components/Calculator.js

```diff
@@ -1,40 +1,139 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import Button from './Button';
 import './calculator.css';
+import calculate from '../logic/calculate';
 
-const Calculator = () => (
-  <div className="calculator-container">
-    <input type="text" value="0" />
-    <div className="buttons-aligner">
-      <Button buttonClass="mainClass" text="AC" />
-      <Button buttonClass="mainClass" text="+/-" />
-      <Button buttonClass="mainClass" text="%" />
-      <Button buttonClass="mainClass subclass" text="&divide;" />
-    </div>
-    <div className="buttons-aligner">
-      <Button buttonClass="mainClass" text="7" />
-      <Button buttonClass="mainClass" text="8" />
-      <Button buttonClass="mainClass" text="9" />
-      <Button buttonClass="mainClass subclass" text="&times;" />
-    </div>
-    <div className="buttons-aligner">
-      <Button buttonClass="mainClass" text="4" />
-      <Button buttonClass="mainClass" text="5" />
-      <Button buttonClass="mainClass" text="6" />
-      <Button buttonClass="mainClass subclass" text="-" />
-    </div>
-    <div className="buttons-aligner">
-      <Button buttonClass="mainClass" text="1" />
-      <Button buttonClass="mainClass" text="2" />
-      <Button buttonClass="mainClass" text="3" />
-      <Button buttonClass="mainClass subclass" text="+" />
-    </div>
-    <div className="buttons-aligner">
-      <Button buttonClass="mainClass subsidiary" text="0" />
-      <Button buttonClass="mainClass" text="." />
-      <Button buttonClass="mainClass subclass" text="=" />
-    </div>
-  </div>
-);
+const Calculator = () => {
+  const [screen, setScreen] = useState('0');
+  const [buttontype, setButtontype] = useState(null);
+  const [calculatoroperation, setCalculatoroperation] = useState(null);
+  const [operationnumber, setOperationnumber] = useState(null);
 
+  useEffect(() => {
+    // console.log(screen, operationnumber, calculatoroperation, buttontype);
+    const total = calculate(
+      {
+        total: screen,
+        next: operationnumber,
+        operation: calculatoroperation,
+      },
+      buttontype,
+    );
+    if (total.total) {
+      setScreen(total.total);
+    } else {
+      setScreen('0');
+    }
+  }, [buttontype, operationnumber]);
+  return (
+    <div className="calculator-container">
+      <input type="text" value={screen} />
```

Comment:
For a good user experience we need to dispaly the operand and operator each time the user clicks a button